src/common/from-hex-did.ts

```diff
@@ -0,0 +1,16 @@
+export const fromHexDID = (did: string) => {
+  const didParts = did.split(':');
+
+  if (didParts.length !== 4) {
+    return did;
+  }
+
```

Comment:
I think this conversation is only relevant for `did:ethr` DIDs as far as I know. I think it could fail for other DID methods, so it's good to have a check for the method here, IMO.
```suggestion

if (didParts[1] !== 'ethr') {
  return did
}

```

Reply:
Not only it must be `ethr` but also DID with chain id

Reply:
Yea, I would keep the `didParts.length !== 4`. I just think if `didParts.length !== 4` is true but `didParts[1] !== 'ethr'` is false, then we probably don't want to process